sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2139,16 +2139,16 @@ class Analyzer(override val catalogManager: CatalogManager)
     }
 
     def lookupBuiltinOrTempFunction(name: Seq[String]): Option[ExpressionInfo] = {
-      if (name.length == 1) {
-        v1SessionCatalog.lookupBuiltinOrTempFunction(name.head)
+      if (name.length == 1 || name.length == 3) {
```

Comment:
why is this change?

Reply:
to look up temp function with `name` = `spark_default.default.and`

Reply:
let me try to handle system functions and user functions seperately, then the changes in `Analyzer` could be reverted.